src/filesystem.cpp

```diff
@@ -309,44 +310,47 @@ AsyncIOContext *FileAccess::asyncfopen(string *f)
 
 bool FileAccess::asyncopenf()
 {
+    numAsyncReads++;
```

Comment:
Yes. The meaning is the one commented by @polmr , to control that `asyncopenf` is called the same number of times than `asyncclosef'  before the deletion of the `FileAccess` object.

To control if the file has been already opened or not, there is a new flag (`isAsyncOpened`)